pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -100,15 +102,35 @@ func setPodInterfaceCache(iface *v1.Interface, podInterfaceName string, uid stri
 		return nil
 	}
 
+	cache := PodCacheInterface{Iface: iface}
+
+	indexIPv4 := 0
+	indexIPv6 := 1
```

Comment:
When there is only a single IP, this logic does not make sense.

Reply:
please explain why not
because you dont think its correct or because there is a way to do the code nicer (taking into account this code handles all the cases - single ipv4 , single ipv6, dual where ipv4 is primary, future dual where ipv6 is primary

Reply:
refactored, please see last commit, once all is fine i will squash. thanks